mono/metadata/threadpool-worker-default.c

```diff
@@ -127,16 +122,15 @@ typedef union {
 typedef struct {
 	MonoRefCount ref;
 
+	MonoThreadPoolWorkerCallback callback;
+
 	ThreadPoolWorkerCounter counters;
 
 	MonoCoopMutex parked_threads_lock;
 	gint32 parked_threads_count;
 	MonoCoopCond parked_threads_cond;
 
-	ThreadPoolWorkItem *work_items; // ThreadPoolWorkItem []
 	gint32 work_items_count;
```

Comment:
Shouldn't this be volatile or read with InterlockedRead, given we're doing atomics with it?